### PR TITLE
Conflict checking enabled by default [DPP-850]

### PR DESCRIPTION
**Pull request by tudor-da**
This PR enables conflict-checking by default in Sandbox-on-X:
* `enable-conflict-checking` CLI flag is removed
* `disable-conflict-checking` CLI flag is added

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/sandbox-on-x/src/main/scala/com/daml/ledger/sandbox/bridge/validate/TagWithLedgerEndImpl.scala [hunk 1]
```diff
@@ -24,7 +24,6 @@ private[validate] class TagWithLedgerEndImpl(
     case Left(rejection) => Future.successful(Left(rejection))
     case Right(preparedSubmission) =>
       Timed.future(
-        // TODO SoX: Remove as this should be instantaneous
```
Comment (tudor-da) [inline ledger/sandbox-on-x/src/main/scala/com/daml/ledger/sandbox/bridge/validate/TagWithLedgerEndImpl.scala:27]: Keeping it for symmetry with the other stages

File: ledger/sandbox-on-x/src/main/scala/com/daml/ledger/sandbox/bridge/LedgerBridge.scala [hunk 2]
```diff
@@ -39,7 +39,6 @@ object LedgerBridge {
       timeProvider: TimeProvider,
   )(implicit
       loggingContext: LoggingContext,
-      // TODO SoX: Consider using a dedicated thread-pool for the ledger bridge
```
Comment (tudor-da) [inline ledger/sandbox-on-x/src/main/scala/com/daml/ledger/sandbox/bridge/LedgerBridge.scala:42]: I've explored using a dedicated thread-pool but with worse performance

**Comment by tudor-da** [comment]
@cocreature I had to disable conflict checking for `create-daml-app`. With it enabled by default the `//:create-daml-app-1.18.1-platform-0.0.0` fails with 
```
io.grpc.StatusRuntimeException: NOT_FOUND: PARTY_NOT_KNOWN_ON_LEDGER(11,6c6e5eef): Parties not known on ledger
```

**Comment by cocreature** [comment]
> I am not sure this is fine. Wdyt?

fine temporarily, Opened https://github.com/digital-asset/daml/issues/12932 to fix it

File: compatibility/bazel_tools/create-daml-app/index.test.ts [hunk 3]
```diff
@@ -138,7 +138,8 @@ beforeAll(async () => {
 
   const sandboxOnXOptions = [
     `--ledger-id=${SANDBOX_LEDGER_ID}`,
-    `--participant=participant-id=sandbox,port=0,port-file=${SANDBOX_PORT_FILE_NAME}`
+    `--participant=participant-id=sandbox,port=0,port-file=${SANDBOX_PORT_FILE_NAME}`,
+    `--disable-conflict-checking`,
```
Comment (cocreature) [inline compatibility/bazel_tools/create-daml-app/index.test.ts:142]:
```suggestion
    // TODO https://github.com/digital-asset/daml/issues/12932
    `--disable-conflict-checking`,
```

**Review by cocreature — APPROVED** [review]

File: compatibility/bazel_tools/create-daml-app/index.test.ts [hunk 4]
```diff
@@ -138,7 +138,9 @@ beforeAll(async () => {
 
   const sandboxOnXOptions = [
     `--ledger-id=${SANDBOX_LEDGER_ID}`,
-    `--participant=participant-id=sandbox,port=0,port-file=${SANDBOX_PORT_FILE_NAME}`
+    `--participant=participant-id=sandbox,port=0,port-file=${SANDBOX_PORT_FILE_NAME}`,
+    // TODO https://github.com/digital-asset/daml/issues/12932
+    `--disable-conflict-checking`,
```
Comment (cocreature) [inline compatibility/bazel_tools/create-daml-app/index.test.ts:143]:
```suggestion
    `--participant=participant-id=sandbox,port=0,port-file=${SANDBOX_PORT_FILE_NAME}`,
```
fixed the issue in #12936

Comment (cocreature) [inline compatibility/bazel_tools/create-daml-app/index.test.ts:143]:
```suggestion
```
fixed in #12936

Comment (tudor-da) [reply compatibility/bazel_tools/create-daml-app/index.test.ts:143]: Ok, will replace

Comment (tudor-da) [reply compatibility/bazel_tools/create-daml-app/index.test.ts]: Thanks @moritzkiefer-da . Done e92e52ff05475f1cbf571504c51bc4e398f1e175

**Review by mziolekda — APPROVED** [review]

**Pull request merged**